File: bwl.Imaging/StructureMatrix/StructureMatrixHSV.vb
Public Class StructureMatrixHSV
    Inherits StructureMatrix(Of HSV)

    Public Sub New(width As Integer, height As Integer)
        MyBase.New(width, height)
    End Sub

    Public Sub New(matrix As HSV(), width As Integer, height As Integer)
        MyBase.New(matrix, width, height)
    End Sub

    Public Sub New(rgbmatrix As RGBMatrix)
        MyBase.New(rgbmatrix.Width, rgbmatrix.Height)
        For i = 0 To _matrix.Length - 1
            _matrix(i) = (New RGB(rgbmatrix.Red(i), rgbmatrix.Green(i), rgbmatrix.Blue(i))).ToHSV
        Next
    End Sub

End Class

File: Bwl.Imaging.Skia/StructureMatrix/StructureMatrixHSV.cs
namespace Bwl.Imaging.Skia;

public class StructureMatrixHSV : StructureMatrix<HSV>
{

    public StructureMatrixHSV(int width, int height) : base(width, height)
    {
    }

    public StructureMatrixHSV(HSV[] matrix, int width, int height) : base(matrix, width, height)
    {
    }

    public StructureMatrixHSV(RGBMatrix rgbmatrix) : base(rgbmatrix.Width, rgbmatrix.Height)
    {
        for (int i = 0, loopTo = _matrix.Length - 1; i <= loopTo; i++)
            _matrix[i] = new RGB(rgbmatrix.Red[i], rgbmatrix.Green[i], rgbmatrix.Blue[i]).ToHSV();
    }

}